tests/testsuite/profile_trim_paths.rs

```diff
@@ -0,0 +1,406 @@
+        .file("Cargo.toml", &basic_manifest("bar", "0.0.1"))
+        .file("src/lib.rs", r#"pub fn f() { println!("{}", file!()); }"#)
+        .build();
+    let bar_path = bar.url().to_file_path().unwrap();
+    let bar_path = bar_path.display();
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [dependencies]
+                bar = { path = "../bar" }
+
+                [profile.dev]
+                trim-paths = "object"
+           "#,
+        )
+        .file("src/main.rs", "fn main() { bar::f(); }")
+        .build();
+
+    p.cargo("run --verbose -Ztrim-paths")
+        .masquerade_as_nightly_cargo(&["-Ztrim-paths"])
+        .with_stdout("bar-0.0.1/src/lib.rs")
+        .with_stderr(
+            &format!("\
+[COMPILING] bar v0.0.1 ([..]/bar)
+[RUNNING] `rustc [..]-Zremap-path-scope=object --remap-path-prefix=[..]/lib/rustlib/src/rust=/rustc/[..] --remap-path-prefix={bar_path}=bar-0.0.1 [..]
+[COMPILING] foo v0.0.1 ([CWD])
+[RUNNING] `rustc [..]-Zremap-path-scope=object --remap-path-prefix=[..]/lib/rustlib/src/rust=/rustc/[..] --remap-path-prefix=[CWD]= [..]
+[FINISHED] dev [..]
+[RUNNING] `target/debug/foo[EXE]`"),
+        )
+        .run();
+}
+
+#[cargo_test(nightly, reason = "-Zremap-path-scope is unstable")]
+fn diagnostics_works() {
+    Package::new("bar", "0.0.1")
+        .file("Cargo.toml", &basic_manifest("bar", "0.0.1"))
+        .file("src/lib.rs", r#"pub fn f() { let unused = 0; }"#)
+        .publish();
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+
+                [dependencies]
+                bar = "0.0.1"
+
+                [profile.dev]
+                trim-paths = "diagnostics"
+           "#,
+        )
+        .file("src/lib.rs", "")
+        .build();
+
+    let registry_src = paths::home().join(".cargo/registry/src");
+    let pkg_remap = format!("{}/[..]/bar-0.0.1=bar-0.0.1", registry_src.display());
+
+    p.cargo("build -vv -Ztrim-paths")
+        .masquerade_as_nightly_cargo(&["-Ztrim-paths"])
+        .with_stderr_contains("\
+warning: unused variable: `unused`
+ --> bar-0.0.1/src/lib.rs:1:18
+  |
+1 | pub fn f() { let unused = 0; }
+  |                  ^^^^^^ help: if this is intentional, prefix it with an underscore: `_unused`
+  |
+  = note: `#[warn(unused_variables)]` on by default",
+        )
+        .with_stderr_contains(
+            &format!("\
+[RUNNING] [..]rustc [..]-Zremap-path-scope=diagnostics --remap-path-prefix=[..]/lib/rustlib/src/rust=/rustc/[..] --remap-path-prefix={pkg_remap} [..]",
+            ))
+        .with_stderr_contains("\
+[RUNNING] [..]rustc [..]-Zremap-path-scope=diagnostics --remap-path-prefix=[..]/lib/rustlib/src/rust=/rustc/[..] --remap-path-prefix=[CWD]= [..]",
+        )
+        .run();
+}
```

Comment:
I'd like to set up a smoke test against debug symbol, so that we can make sure they are sanitized. I can follow [what rustc does](https://github.com/Urgau/rust/blob/8501f1c7ba440c97833ea9dec03965b4d5ef2fd7/tests/run-make/split-debuginfo/Makefile#L114-L121) by using `objdump`. Is there any better way to verify it? What about cross-platform tests?

Reply:
Ended up using `readelf`. Now we have covered Linux for `object` option. I think that's pretty acceptable at this stage. We can add more tests for macOS and Windows later.